rfcs/0003-languages.md

```diff
@@ -0,0 +1,159 @@
+---
+feature: (languages)
+start-date: (2021-04-30)
+author: (Shahrukh Khan)
+related-issues: (None)
+---
+
+# Summary
+[summary]: #summary
+
+This RFC describes how language support is to be implemented in Soxin. This
+breaks down in two areas:
+
+* How languages are defined, i-e how to define the language and its properties
+  i-e its compiler, tools, editor plugins etc. How the user can define the
+  defualt plugins for a language in his editor(s).
+* How the user can select default set of languages and its required tools in
+  his editor, and how to override default settings on the editor.
+
+# Motivation
+[motivation]: #motivation
+
+As Soxin is aimed at providing its users with easy-to-configure software, while
+allowing for great customization, an RFC describing how such an important
+feature integrates with the project was needed.
+
+# Detailed design
+[design]: #detailed-design
+
+## Language definition
+
+Languages would be defined in a module named `soxin.programming.languages`.
+Each language, would be an attribute set in this module:
+
+```
+soxin.programming.languages = {
+  go = { /* ... */ };
+  java = { /* ... */ };
+  python = { /* ... */ };
+}
+```
+## Tools defination
+
+Programming tools will be defined in a module named `soxin.programming.tools`.
```

Comment:
If I recall correctly, we discussed the name of this option with @kalbasit two weeks ago, and concluded it would be better named `soxin.tools`, as some of the tools we would put in there are not only used for programming.
We might also consider renaming `soxin.programming.languages` to `soxin.programmingLanguages`, along with the appropriate options in `soxin.settings` and per-module, of course.